MMM-Tibber.js

```diff
@@ -235,51 +235,43 @@ Module.register("MMM-Tibber", {
       return;
     }
-    if (notification === "TIBBER_SUBSCRIPTION_DATA") {
-      // this.log("Got sub data: ");
-      // this.log(payload);
-      this.updateSubData(payload);
-    } else if (notification === "TIBBER_DATA") {
-      this.log("Got Tibber data: ");
-      this.log(payload);
-      const additionalCosts = this.sumAdditionalCosts(this.config);
-      // this.makeSavingsData(payload);
-      if (this.config.showConsumption || this.config.showPrice) {
-        clearInterval(this.interval);
-        drawGraphs(
-          this.identifier,
-          new TibberData(payload),
-          this.config,
-          additionalCosts,
-          this.savingsData
-        );
-        this.interval = setInterval(() => {
+
+    const subscription_prefix = "TIBBER_SUBSCRIPTION_DATA_";
+    const data_prefix = "TIBBER_DATA_";
+
+    if (notification.startsWith(subscription_prefix)) {
+      const instanceId = notification.substring(subscription_prefix.length);
+      if (instanceId == this.instanceId) {
+        this.log("got subscription data for instance: " + instanceId + " current instance is: " + this.instanceId);
+        this.updateSubData(payload);
+      }
+    } else if (notification.startsWith(data_prefix)) {
+      const instanceId = notification.substring(data_prefix.length);
+      if (instanceId == this.instanceId) {
+        this.log("got usage data for instance: " + instanceId + " current instance is: " + this.instanceId);
+        this.log("Got Tibber data: ");
+        this.log(payload);
+        const additionalCosts = this.sumAdditionalCosts(this.config);
+        // this.makeSavingsData(payload);
+        if (this.config.showConsumption || this.config.showPrice) {
+          clearInterval(this.interval);
           drawGraphs(
             this.identifier,
             new TibberData(payload),
             this.config,
             additionalCosts,
             this.savingsData
           );
-        }, 30000);
-      }
-    }
-  },
-
-  notificationReceived: function (notification, payload, sender) {
-    if (notification === "MQTT_MESSAGE_RECEIVED") {
-      console.log("MMM-Tibber received MQTT_MESSAGE_RECEIVED notification");
-      console.log({ payload, sender });
-      if (payload.topic === this.config.savingsTopic) {
-        const savings = JSON.parse(payload.value);
-        const sumAdditional = this.config.showAdditionalCostsGraph
-          ? this.sumAdditionalCosts(this.config)
-          : 0;
-        this.savingsData = savings.hours.map((h) => [
-          Date.parse(h.start),
-          h.saving ? sumAdditional + h.price - h.saving : null,
-          h.saving ? sumAdditional + h.price : null
-        ]);
+          this.interval = setInterval(() => {
+            drawGraphs(
+              this.identifier,
+              new TibberData(payload),
+              this.config,
+              additionalCosts,
+              this.savingsData
+            );
```

Comment:
Good idea to call this immediately in addition to the interval, but I suggest you wrap these 7 lines in a function with no arguments so only one line needs to be duplicated. Tjat makes it safer in case the calls must be changed.

Reply:
This is already a function call. I did not like the code after trying this refactor. It became hard to read and reason. Better to have the function all duplication, in my eyes. Happy to see another attempt, but mine was not nice.